Box.V2/Managers/BoxFilesManager.cs

```diff
@@ -646,5 +646,52 @@ private int GetTimeDelay(HttpResponseHeaders headers)
 
             return response.ResponseObject;
         }
+
+        /// <summary>
+        /// Discards a specific file version to the trash.
+        /// </summary>
+        /// <param name="id">Id of the file (Required).</param>
+        /// <param name="versionId">Id of the version (Required).</param>
+        /// <param name="etag">The etag of the file. This is in the ‘etag’ field of the file object.</param>
+        /// <returns>True, if version is deleted..</returns>
+        public async Task<bool> DeleteOldVersionAsync(string id, string versionId, string etag = null)
+        {
+            id.ThrowIfNullOrWhiteSpace("id");
+            versionId.ThrowIfNullOrWhiteSpace("versionId");
+
+            BoxRequest request = new BoxRequest(_config.FilesEndpointUri, string.Format(Constants.DeleteOldVersionPathString, id, versionId))
+                .Method(RequestMethod.Delete)
+                .Header(Constants.RequestParameters.IfMatch, etag);
+
+            IBoxResponse<BoxFile> response = await ToResponseAsync<BoxFile>(request).ConfigureAwait(false);
+
+            return response.Status == ResponseStatus.Success;
+        }
+
+        /// <summary>
+        /// If there are previous versions of this file, this method can be used to promote one of the older versions to the top of the stack. 
+        /// This actually mints a copy of the old version and puts it on the top of the versions stack. 
+        /// The file will have the exact same contents, the same SHA1/etag, and the same name as the original. 
```

Comment:
The file will have the exact same contents, the same SHA1/etag, and the same name as the original. 

I am thinking "as the promoted version"

@cburnette do we need to update our docs.box.com also?